cirq/experiments/fidelity_estimation.py

```diff
@@ -372,3 +394,225 @@ def least_squares_xeb_fidelity_from_probabilities(
+    with ThreadPoolExecutor(max_workers=2) as pool:
+        futures = [pool.submit(run_batch, task) for task in b_tasks]
+
+        records = []
+        with tqdm.tqdm(total=n_tasks) as progress:
+            for future in concurrent.futures.as_completed(futures):
+                records += future.result()
+                progress.update(batch_size)
+
+    return pd.DataFrame(records).set_index(['circuit_i', 'cycle_depth'])
+
+
+def _simulate_2q_xeb_circuit(task: Dict[str, Any]):
+    """Helper function for simulating a given (circuit, cycle_depth)."""
+    circuit_i = task['circuit_i']
+    cycle_depth = task['cycle_depth']
+    circuit = task['circuit']
+    param_resolver = task['param_resolver']
+
+    circuit_depth = cycle_depth * 2 + 1
+    assert circuit_depth <= len(circuit)
+    tcircuit = circuit[:circuit_depth]
+    tcircuit = protocols.resolve_parameters_once(tcircuit, param_resolver=param_resolver)
+
+    pure_sim = sim.Simulator()
+    psi = cast(sim.StateVectorTrialResult, pure_sim.simulate(tcircuit))
+    psi = psi.final_state_vector
+    pure_probs = np.abs(psi) ** 2
+
+    return {
+        'circuit_i': circuit_i,
+        'cycle_depth': cycle_depth,
+        'pure_probs': pure_probs,
+    }
+
+
+def simulate_2q_xeb_circuits(
+    circuits: Sequence['cirq.Circuit'],
+    cycle_depths: Sequence[int],
+    param_resolver: 'cirq.ParamResolverOrSimilarType' = None,
+    pool: Optional['multiprocessing.pool.Pool'] = None,
+):
+    """Simulate two-qubit XEB circuits.
+
+    These ideal probabilities can be benchmarked against potentially noisy
+    results from `sample_2q_xeb_circuits`.
+
+    Args:
+        circuits: A library of two-qubit circuits generated from
+            `random_rotations_between_two_qubit_circuit` of sufficient length
+            for `cycle_depths`.
+        cycle_depths: A sequence of cycle depths at which we will truncate each
+            of the `circuits` to simulate.
+        param_resolver: If circuits contain parameters, resolve according
+            to this ParamResolver prior to simulation
+        pool: If provided, execute the simulations in parallel.
+
+    Returns:
+        A dataframe with index ['circuit_i', 'cycle_depth'] and column
+        "pure_probs" containing the pure-state probabilities for each row.
+    """
+    tasks = []
+    for cycle_depth in cycle_depths:
+        for circuit_i, circuit in enumerate(circuits):
+            tasks += [
+                {
+                    'circuit_i': circuit_i,
+                    'cycle_depth': cycle_depth,
+                    'circuit': circuit,
+                    'param_resolver': param_resolver,
+                }
+            ]
+
+    if pool is not None:
+        records = pool.map(_simulate_2q_xeb_circuit, tasks, chunksize=4)
+    else:
+        records = [_simulate_2q_xeb_circuit(record) for record in tasks]
+
+    return pd.DataFrame(records).set_index(['circuit_i', 'cycle_depth'])
+
+
+def simulate_2q_xeb_fidelities(
```

Comment:
Is simulate a correct verb here? benchmark_2q_xeb_fidelities?

Reply:
simulate is not the correct verb. Benchmark is better. I guess I was going for the idea that the performance of this function is limited by simulating the circuits.

`simulate_in_order_to_benchmark_2q_xeb_circuits` is too long of a name :) Changing to `benchmark_2q_xeb_fidelities`.